src/nwb_benchmarks/core/_base_network_benchmark.py

```diff
@@ -0,0 +1,102 @@
+"""Base class for implementing benchmarks for evaluating network performance metrics for streaming read of NWB files."""
+
+import os
+import time
+
+from asv_runner.benchmarks.mark import SkipNotImplemented
+
+from ._capture_connections import CaptureConnections
+from ._network_profiler import NetworkProfiler
+from ._network_statistics import NetworkStatistics
+
+
+class BaseNetworkBenchmark:
+    """
+    Base class for ASV Benchmark network performance metrics for streaming data access.
+
+    Child classes must implement:
+        1) The `operation_to_track_network_activity_of`, which replaces the usual time_ or mem_ operation being tracked.
+        2) The `setup_cache` method to explicitly recieve the necessary keyword arguments to be passed to the operation.
+    """
+
+    def operation_to_track_network_activity_of(self, **keyword_arguments):
+        raise SkipNotImplemented()
+
+    def setup_cache(self, **keyword_arguments) -> dict:
+        """
+        Strategy essentially mimics a single benchmark execution (setup + run).
+
+        Only supports single operations (ignores `repeat` and prohibits `params`).
+        """
+        self.start_net_capture()
+        t0 = time.time()
+        self.setup(**keyword_arguments)
```

Comment:
As far as I understand the docs of ASV, the `setup` methods is always being execute for a test and `setup_cache` is executed only once. I.e., I think even if a class has a `setup_cache` methods, ASV will still call the `setup` method as well. In this case, we don't want the `setup` method to be called again. As such, I think we should use another method name for this here to allow for a separate setup for `self.operation_to_track_network_activity_of`

Also, the setup method should be called at the beginning of the function, before the timer and `self.start_net_capture()`, as optherwise the setup is being profiled together with the `self.operation_to_track_network_activity_of`, and I think we here only want to profile `self.operation_to_track_network_activity_of` by itself.

Reply:
See https://github.com/NeurodataWithoutBorders/nwb_benchmarks/pull/22 for other proposals that simplify this approach even more

> I.e., I think even if a class has a setup_cache methods, ASV will still call the setup method as well.

Correct. `setup_cache` called one, `setup` called many